lib/yamllint/linter.rb

```diff
@@ -43,10 +43,20 @@ def check(path)
         end
       end
 
-      File.open(path, 'r') do |f|
-        error_array = []
-        valid = check_data(f.read, error_array)
-        errors[path] = error_array unless error_array.empty?
+      commentcount = false
+      File.readlines(path).each do |line|
+        if(!(line.include? "#"))
```

Comment:
Travis did not run but I think this will fail rubocop since you use if not instead of unless

Reply:
changed to unless